datafusion/functions/src/datetime/to_timestamp.rs

```diff
@@ -561,6 +561,31 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn to_timestamp_with_invalid_tz() -> Result<()> {
```

Comment:
I think the slt coverage is sufficient, so this test is not necessary. However, I don't think that would stop this PR from merging.